### PR TITLE
fix(codegen): required param error implementation

### PR DESCRIPTION
**Pull request by Karitham**
- return correctly required param
- delete dead code
- replace usage of deprecated function
- update go:generate command of moq to use latest, since it seems to be
  causing issues.
- update required param error tests

POSSIBLE BREAKING: multiple word call to `title` helper function (no
change in generation, and it generates valid code, but to be aware if
needs be).

### CODE REVIEW

File: internal/test/server/doc.go
Comment (Karitham) [inline internal/test/server/doc.go:4]: I'm not sure why I can't go generate without that, but I need @latest for it to pass.
If it works on your end I could remove it, or even tag another version

**Comment by diamondburned** [comment]
I created PR https://github.com/discord-gophers/goapi-gen/pull/85 which depends on this PR before it can be merged.

**Comment by Karitham** [comment]
Waiting for review from other maintainer before merging this one. Will be merging both then

**Review by hhhapz — APPROVED** [review]

**Pull request merged**
